cmd/ncproxy/main.go

```diff
@@ -0,0 +1,117 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"github.com/Microsoft/hcsshim/cmd/ncproxy/nodenetsvc"
+	"github.com/Microsoft/hcsshim/internal/computeagent"
+	"github.com/Microsoft/hcsshim/internal/log"
+	"google.golang.org/grpc"
+)
+
+type nodeNetSvcConn struct {
+	client   nodenetsvc.NodeNetworkServiceClient
+	addr     string
+	grpcConn *grpc.ClientConn
+}
+
+var (
+	configPath = flag.String("config", "", "Path to JSON configuration file.")
+	// Global mapping of network namespace ID to shim compute agent ttrpc service.
+	containerIDToShim = make(map[string]computeagent.ComputeAgentService)
+	// Global object representing the connection to the node network service that
+	// ncproxy will be talking to.
+	nodeNetSvcClient *nodeNetSvcConn
+	// NcProxy configuration
+	conf = new(config)
```

Comment:
I'd really rather we just pass config info to the functions that need it rather than using this as a global across multiple files.

Reply:
Yea agreed looking again.